src/operations/indexes.ts

```diff
@@ -212,16 +245,12 @@ export class CreateIndexesOperation extends CommandOperation<string[]> {
 
     this.options = options ?? {};
     this.collectionName = collectionName;
-    this.indexes = indexes.map(userIndex => {
+    this.indexes = indexes.map((userIndex: IndexDescription): ResolvedIndexDescription => {
       // Ensure the key is a Map to preserve index key ordering
       const key =
         userIndex.key instanceof Map ? userIndex.key : new Map(Object.entries(userIndex.key));
       const name = userIndex.name != null ? userIndex.name : Array.from(key).flat().join('_');
```

Comment:
Optional, but while we're here
```suggestion
      const name = userIndex.name ?? Array.from(key).flat().join('_');
```